pkg/vcs/vcs.go

```diff
@@ -58,3 +85,117 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App, cloneDir string) (string, string, error) {
+	vcs := app.Status.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" || vcs.BuildContext == "" {
+		return "", "", fmt.Errorf("app is missing required vcs information, image must be rebuilt with a newer acorn cli")
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote)-1 {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Join(cloneDir, strings.TrimSuffix(remote[idx+1:], ".git"))
+
+		// Check for the directory we want to use
+		f, err := os.Open(workdir)
+		if err == nil {
+			// Directory exists, check if empty
+			_, err = f.ReadDir(1)
+			if err != nil {
+				// Directory is empty, clone the repo
+				err = gitClone(ctx, workdir, remote)
+				if err != nil {
+					fmt.Printf("%v\n", err)
+					continue
+				}
+			} else {
+				// Directory is not empty, try to fetch
+				err = gitFetch(ctx, workdir, remote)
+				if err != nil {
+					fmt.Printf("%v\n", err)
+					continue
+				}
+			}
+		} else if os.IsNotExist(err) {
+			// Directory does not exist, just clone to create it
+			err = gitClone(ctx, workdir, remote)
+			if err != nil {
+				fmt.Printf("%v\n", err)
+				continue
+			}
+		} else {
+			fmt.Printf("failed to check for the existence of directory %q: %v", workdir, err)
+			continue
+		}
+
+		// Try to checkout the revision
+		err = gitCheckout(ctx, workdir, vcs.Revision)
+		if err != nil {
+			fmt.Printf("%v\n", err)
+			continue
+		}
+
+		// Create the Acornfile in the repository
+		acornfile := filepath.Join(workdir, vcs.Acornfile)
+		err = os.WriteFile(acornfile, []byte(app.Status.AppImage.Acornfile), 0666)
+		if err != nil {
+			fmt.Printf("failed to create file %q in repository %q: %v\n", acornfile, workdir, err)
+			continue
+		}
+
+		// Determine if the Acornfile is dirty or not
+		if gitDirty(ctx, workdir) {
+			fmt.Printf("running with a dirty Acornfile %q\n", acornfile)
+		}
+
+		// Get the build context
+		buildContext := filepath.Join(workdir, vcs.BuildContext)
+
+		return acornfile, buildContext, nil
+	}
+	return "", "", fmt.Errorf("failed to resolve an acornfile from the app")
+}
+
+func gitClone(ctx context.Context, workdir, remote string) (err error) {
+	args := []string{"clone", remote, workdir}
+	cmd := exec.CommandContext(ctx, "git", args...)
+	cmdErr := cmd.Run()
+	if cmdErr != nil {
+		err = fmt.Errorf("failed to clone repository %q: %v", remote, err)
+	}
+	return
+}
+
+func gitFetch(ctx context.Context, workdir, remote string) (err error) {
+	args := []string{"-C", workdir, "fetch", remote}
+	cmd := exec.CommandContext(ctx, "git", args...)
+	cmdErr := cmd.Run()
```

Comment:
does this get the command's error from stderr?

Reply:
```If the command starts but does not complete successfully, the error is of type *ExitError. Other error types may be returned for other situations.```
I don't believe so, I could go in and route the stderr to the normal stderr so that the output still gets generated and we get output in the shell, I'll need to investigate this.

Reply:
Yeah I just routed all the outputs to the terminal, since that's what we would do with it anyways